cli/azd/pkg/input/console.go

```diff
@@ -192,6 +219,36 @@ func (c *AskerConsole) MessageUxItem(ctx context.Context, item ux.UxItem) {
 	}
 }
 
+func defaultShowPreviewerOptions() *ShowPreviewerOptions {
+	return &ShowPreviewerOptions{
+		MaxLineCount: 5,
+	}
+}
+
+func (c *AskerConsole) ShowPreviewer(ctx context.Context, options *ShowPreviewerOptions) *ConsolePreviewer {
```

Comment:
Is it possible to have the V1 implementation treat `progressLog` as a magical standalone component not tied to `console`, even though the real-world implications is that you only have a single terminal buffer? Something about the abstractions here don't quite feel right. Having the standalone component might not matter if we have a world of single-threaded execution

Something about `ShowPreviewer()` which changes the `console` instance behavior for `Message` and `ShowSpinner` doesn't quite sit right...

Reply:
We can try that approach as part of getting rid of async. 
But even in a sync world, we can't assume that go-routines would want to communicate things that are happening at the same time. For example, during provisioning and showing the progress.